KK_ButtPlugin/ButtplugController.cs

```diff
@@ -198,12 +189,12 @@ private float GetSpeedMultiplierFor(float animStrokeTimeSecs)
             return Math.Min(1, animStrokeTimeSecs * ButtPlugin.MaxStrokesPerMinute.Value / 60f);
         }
 
-        private void DoStroke(int strokeTimeMs, double margin)
+        IEnumerator DoStroke(int strokeTimeMs, double margin)
         {
             client.LinearCmd(
                 position: 1 - margin * 0.7,
                 durationMs: strokeTimeMs / 2);
-            Thread.Sleep(strokeTimeMs / 2);
+            yield return new WaitForSeconds(strokeTimeMs / 2000);

```

Comment:
lol ouch 😬 forgot that this would result in integer division.  I unfortunately don't have a stroker capable device to test with, so I can't test regressions

Reply:
no worries, that's what I'm here for